cmd/pebble/cmd_run.go

```diff
@@ -139,6 +141,9 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal) error {
 	if err := d.Init(); err != nil {
```

Comment:
It feels a little bit weird to call Init() here, which creates all the socket listeners, only to return and exit knowing those will never be used.

Would it not be cleaner to call d.Validate() in runDaemon() before we call d.Init(), and only proceed to Init() if not --dry?

Reply:
We've removed the `Dry` flag from the daemon, but I agree with your concern here. What about we explicitly call `d.Validate()` in `cmd_run.go` above the `d.Init()` call, then check `rcmd.Dry` after that, then call `d.Init()` (the plan is cached, so that's fine to do in terms of efficiency).